### PR TITLE
Evaluate OSS for Security

### PR DESCRIPTION
**Pull request by balteravishay**
# Pull Request Template

## What are you trying to address

This PR addresses issue #1039 by proposing a set of checks and tools that can be applied when evaluating an OSS package.

## Checklist

[READY TO PR? Use the check-list below to ensure your branch is ready for PR.]

- [v] Changes follow the repo structure and land in the appropriate folder and section
- [v] No confidential information
- [v] No duplicated content
- [v] Labeled appropriately
- [v] Added 2 reviewers
- [v] No lint errors
- [v] No lint check errors related to your changes
      
> Note: You may see link check errors on pages you have not touched. This is normal, and due to either broken links or sites that reject link checker bots. The reviewer will help you get to a green state on these.

### CODE REVIEW

**Review by nyouens — APPROVED** [review]
Looks good

**Review by anatbal — APPROVED** [review]
Looks really good and super important!

**Comment by superhindupur** [comment]
Thanks for adding this, very helpful. As discussed on the EMEA security committee call, it would be great if we can add some guidance to check for the dependency tree of a new open source dependency during a code review.

**Comment by balteravishay** [comment]
thanks @superhindupur, great feedback! added a section on "When to evaluate OSS", wdyt?

File: docs/continuous-integration/dev-sec-ops/evaluate-oss/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+- Can you avoid adding it as a dependency? The best dependency is the one you don't have.
+- Is it maintained? How often and at what engineering rigor (i.e. code reviews, branch protection, tests)
+- Is there evidence that effort is taken to make it secure?
+- Can you find a reference that it is used significantly downstream by other projects or is referenced by known and trusted documentation?
+- Is it easy to use securely?
+- Are there instructions on how to report vulnerabilities? 
+- Does it have any known vulnerabilities or security issues?
+- Are its dependencies secure, or at least up to date and actively maintained?
+- Has it been audited by a third party such as the [OpenSSF Security Reviews](https://github.com/ossf/security-reviews/blob/main/Overview.md#readme)?
```
Comment (shiranr) [inline docs/continuous-integration/dev-sec-ops/evaluate-oss/README.md:28]: I would add to check the license of the software, how many stars and forks it has.

File: docs/continuous-integration/dev-sec-ops/evaluate-oss/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Evaluate Open Source Software
+
+Identify potential candidates for open-source dependencies and evaluate them against your needs and the required security posture.
+
+## Why Evaluate Open Source Software
```
Comment (shiranr) [inline docs/continuous-integration/dev-sec-ops/evaluate-oss/README.md:5]: I would consider adding this as examples:
https://david-gilbertson.medium.com/im-harvesting-credit-card-numbers-and-passwords-from-your-site-here-s-how-9a8cb347c5b5

Comment (balteravishay) [reply docs/continuous-integration/dev-sec-ops/evaluate-oss/README.md:5]: added another one (Microsoft one :) )

**Review by shiranr — APPROVED** [review]

**Pull request merged**